reesegreen2014/TaskPro
# Feature/cypress testing

#### What does this PR do?
- Implements Cypress testing for the TaskPro application.
- Updates the "edit task" functionality to ensure tasks can be edited properly and changes are reflected in the Redux store and the UI.
- Introduces Cypress custom commands to streamline the testing process.
#### Where should the reviewer start?
- Review the Cypress tests located in the cypress directory to understand the tests that have been implemented.
- Check the cypress/support/commands.js file to see the custom commands created for the Cypress tests.
- Examine the updates in TaskDetailPage.js to see the changes made to the edit task functionality.
#### How should this be manually tested?
- Pull down changes from this branch using `git pull origin feature/cypress-testing`
- Launch cypress using `npx cypress open`
- Review all tests and ensure they are passing
- Review `commands.js` within Cypress directory to understand implementation of commands and how they are called to throughout test file
- Test editing a task, observing the state changes upon update (Ensure you have Redux DevTools extension installed in Google Chrome)
#### Any background context you want to provide?
- This branch focuses on adding comprehensive end-to-end testing using Cypress to ensure the application's functionalities are working as expected.
- The update to the edit task functionality was necessary to ensure data consistency and a smooth user experience.
#### What are the relevant tickets (if any)?
- Relevant issue for User Story #18 will be closed upon successful merge.
#### Screenshots (if appropriate)
![image](https://github.com/user-attachments/assets/7ba2b97c-4feb-4623-9ec9-6c2252312f82)
![image](https://github.com/user-attachments/assets/c10e653c-d780-407f-ab87-46d6ac8737ca)
